src/storage/state.rs

```diff
@@ -0,0 +1,151 @@
+use std::collections::HashMap;
+
+use derive_builder::Builder;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+
+use crate::events::Linkdef;
+
+////////////
+// Claims //
+////////////
+
+/// A Claim that can be applied to a given lattice
+#[derive(Debug, Serialize, Deserialize, Clone, Builder, PartialEq, Eq)]
+#[builder(field(public))]
+pub struct Claim {
+    /// Name of the claim
+    pub name: String,
+
+    /// Version of the claim (semver)
+    pub version: Version,
+
+    /// Revision of the claim
+    pub revision: usize,
+
+    /// Subscriber public (n)key
+    pub subscriber: String,
+
+    /// Issuer public (n)key
+    pub issuer: String,
+
+    /// Call alias that can be used for the claim (if present)
+    pub call_alias: Option<String>,
+
+    /// Capabilities that this claim uses
+    #[serde(rename = "caps")]
+    pub capabilities: Vec<String>,
+
+    /// Tags that have been set on the claim
+    pub tags: HashMap<String, ()>,
+}
+
+///////////////
+// Providers //
+///////////////
+
+/// A wasmCloud Capability provider
+#[derive(Debug, Serialize, Deserialize, Clone, Builder, PartialEq, Eq)]
+#[builder(field(public))]
+pub struct Provider {
+    /// ID of the provider, normally a public (n)key
+    pub id: String,
+
+    /// Name of the provider
+    pub name: String,
+
+    /// Issuer of the (signed) provider
+    pub issuer: String,
+
+    /// Contract ID
+    pub contract_id: String,
+
+    /// Tags set on the provider
+    pub tags: Vec<String>,
+}
+
+////////////
+// Actors //
+////////////
+
+#[derive(Debug, Serialize, Deserialize, Clone, Builder, PartialEq, Eq)]
+#[builder(field(public))]
+pub struct Actor {
+    /// ID of the actor, normally a public (n)key
+    pub id: String,
+
+    /// Name of the provider
+    pub name: String,
+
+    /// Capabilities that the actor requires
+    pub capabilities: Vec<String>,
+
+    /// Issuer of the (signed) actor
+    pub issuer: String,
+
+    /// Tags set on the provider
+    pub tags: Vec<String>,
+
+    /// Call alias to use for the actor
+    pub call_alias: String,
+}
+
+////////////
+// Hosts //
+////////////
+
+/// A wasmCLoud host
+#[derive(Debug, Serialize, Deserialize, Clone, Builder, PartialEq, Eq)]
+#[builder(field(public))]
+pub struct Host {
+    /// ID of the host
+    pub id: String,
+
+    /// Name (human-friendly) of the provider
+    pub name: String,
+
+    /// Host Uptime
+    pub uptime_seconds: u64,
+
+    /// Labels set on the provider
+    pub labels: Vec<String>,
+
+    /// Version of the host
+    pub version: Version,
+}
+
+/////////////
+// Lattice //
+/////////////
+
+/// Lattice-wide parameters
+#[derive(Debug, Serialize, Deserialize, Clone, Default, PartialEq, Eq)]
+pub struct LatticeParameters {
+    host_status_decay_rate_seconds: Option<u128>,
```

Comment:
Is this here to match a value we need elsewhere? `u128` is an odd one to see in a public API so I just wanted to check

Reply:
I saw this as a lattice parameter somewhere in some rust code, but will make this a u32!

Reply:
This has been changed to `u32`